lib/elixir/lib/supervisor.ex

```diff
@@ -616,18 +616,35 @@ defmodule Supervisor do
           optional(:significant) => boolean()
         }
 
+  @typedoc """
+  A module-based child spec.
+
+  This is a form of child spec that you can pass to functions such as `child_spec/2`,
+  `start_child/2`, and `start_link/2`.
+
+  A module-based child spec can be:
+
+    * a **module** — the supervisor calls `module.child_spec([])` to retrieve the
+      child specification
+
+    * a **two-element tuple** in the shape of `{module, arg}` — the supervisor
+      calls `module.child_spec(arg)` to retrieve the child specification
+
+  """
+  @typedoc since: "1.16.0"
+  @type module_spec :: {module(), args :: term()} | module()
+
   @doc """
   Starts a supervisor with the given children.
 
   `children` is a list of the following forms:
 
-    * a [child specification](`t:child_spec/0`)
+    * a child specification (see `t:child_spec/0`)
 
-    * a module, where `module.child_spec([])` will be invoked to retrieve
-      its child specification
+    * a module-based child specification (see `t:module_spec/0`)
```

Comment:
Can we avoid the hop in here? Let's briefly mention what it is and then link to the type.